pkg/network/node/iptables.go

```diff
@@ -57,6 +57,14 @@ func (n *NodeIPTables) Setup() error {
 	return nil
 }
 
+// FinishSetup performs bootstrap action that must be made after kube-proxy
+// initialization
```

Comment:
This won't work. Kube-proxy may recreate its rules later, in various circumstances. There's no way to reliably guarantee that your rule gets inserted before kube-proxy's rule.

You may be able to prepend a rule to the beginning of kube-proxy's chain though, assuming it's not one of the chains that gets recreated from scratch on each proxy sync?

Reply:
Does that include the rules in the PREROUTING chain? I see children of KUBE-SERVICES: 
https://github.com/openshift/sdn/blob/master/vendor/k8s.io/kubernetes/pkg/proxy/iptables/proxier.go#L368-L403

However, I can't find anywhere where KUBE-SERVICES  is ever deleted outside the function CleanupLeftovers.

Reply:
The exact details of the code may change in the future. We've learned that you should not try to depend on a particular rule always ending up before or after kube's rules, or it will eventually break.